lib/index.js

```diff
@@ -165,11 +144,12 @@ module.exports = {
     }
 
     this._jqueryIntegrationEnabled =
-      optionalFeaturesMissing || optionalFeatures.isFeatureEnabled('jquery-integration');
+      !optionalFeaturesMissing && optionalFeatures.isFeatureEnabled('jquery-integration');
 
     if (this._jqueryIntegrationEnabled) {
-      this.ui.writeWarnLine(
-        'Setting the `jquery-integration` optional feature flag to `true`, or not providing a setting at all, has been deprecated. You must add the `@ember/optional-features` addon and set this feature to `false`. This warning will become an error in Ember 4.0.0.\n\nFor more information, see the deprecation guide: https://deprecations.emberjs.com/v3.x/#toc_optional-feature-jquery-integration'
+      const SilentError = require('silent-error');
+      throw new SilentError(
```

Comment:
This is the right thing to do? Or should we assert if anyone *sets* this feature enabled/disabled at all? As in, they should just remove it from their configuration.

The approach of require `false` is certainly more gentle, but a new 4.0 user would probably look curiously at a requirement to set a feature to disabled which... could not possibly be enabled.

Reply:
Hm, right, I actually meant to only throw when this is explicitly set to true. I (wrongly) assumed this is already false by default, but [it is not](https://github.com/emberjs/ember-optional-features/blob/master/features/jquery-integration.js#L7). So I will need to revisit this, to make it match [RFC705 Phase 2](https://emberjs.github.io/rfcs/0705-deprecate-jquery-optional-feature.html#phase-2). So I would basically use [`isFeatureExplicitlySet`](https://github.com/emberjs/ember-optional-features/blob/master/index.js#L71) to check if is explicitly set to true, or do noting (optional feature has no effect) otherwise, right?

Reply:
I have addressed this now!